csharp/ql/lib/semmle/code/csharp/commons/Collections.qll

```diff
@@ -66,15 +68,13 @@ class CollectionType extends RefType {
   }
 }
 
-/** An IEnumerable type. */
-class IEnumerableType extends RefType {
-  IEnumerableType() {
-    this.hasQualifiedName("System.Collections", "IEnumerable")
-    or
-    this.(ConstructedType)
-        .getUnboundGeneric()
-        .hasQualifiedName("System.Collections.Generic", "IEnumerable<>")
-  }
+/** Holds if `t` is a collection type. */
+predicate isCollectionType(ValueOrRefType t) {
+  not t instanceof StringType and
+  exists(ValueOrRefType base | base = t.getABaseType*() |
+    base instanceof SystemCollectionsGenericIEnumerableTInterface or
```

Comment:
This line should not be needed, since [`IEnumerable<T>`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.ienumerable-1?view=net-6.0) extends `IEnumerable`.